xarray_beam/_src/core.py

```diff
@@ -430,7 +432,8 @@ def _key_to_chunks(self, key: Key) -> Iterator[Tuple[Key, DatasetOrDatasets]]:
     results = []
     for ds in self._datasets:
       dataset = ds if key.vars is None else ds[list(key.vars)]
```

Comment:
@shoyer Currently, for the case where the first ds has more dimensions and we split (test case on line 470), this line fails. Specifically, the `key.vars` will index a variable that doesn't exist in the dataset. 

I think I've forgotten the solution we spoke about in VC this morning. Any ideas for how we can address this? Did I understand the new constraint correctly?